### ISSUE TITLE
Crashing After webOS 24 (9.x) Upgrade

### ISSUE THREAD

**Issue by gusdom7** [mariotaku/moonlight-tv#461]
# Short Answer by @mariotaku: You need to disable dev mode once (reboot is required), or perform a factory reset, as this is the only way to get apps like Moonlight, Kodi, etc to work. See [Known Issues](https://github.com/mariotaku/moonlight-tv/wiki/Known-Issues#crashing-on-webos-24-after-firmware-upgrade) & https://github.com/mariotaku/moonlight-tv/issues/461#issuecomment-2423922280

---

### Moonlight Version

1.6.20

### Device Type

webOS TV

### Device Model

LG C3

### Device OS

9.2.0 FW23.20.50

### Host OS

Windows

### Host Software

GeForce Experience

### Bug Description

Starting moonlight crashes on startup logo.
Attached is crash log
[crashlog.txt](https://github.com/user-attachments/files/17369704/crashlog.txt)

**Comment by RedGuilhas** [comment]
Happening to me as well. Also LG C3 with the new WebOS version.

**Comment by mariotaku** [comment]
@RedGuilhas Can you provide the crash log too? I'm trying to find which piece caused this.

**Comment by mariotaku** [comment]
@gusdom7 Thanks for the log. Are you having 1.6.18 actually?

**Comment by RedGuilhas** [comment]
Sorry for being a noob but how do I get access to the logs?
I'm on 1.6.20 btw

**Comment by mariotaku** [comment]
@RedGuilhas How did you install the app? Do you mind if we have a chat on Discord, I'd like to investigate as this might be a new compatibility issue

**Comment by RedGuilhas** [comment]
I installed it through the homebrew app. Add me on Discord @guilhas88

**Comment by gusdom7** [comment]
> @gusdom7 Thanks for the log. Are you having 1.6.18 actually?

Tried several versions, just in case... installed via ipk with dev manager.

**Comment by Archimedes1986** [comment]
Same here on LG C3 65" after WebOS 24 Update. Crashes with 1.6.20 and the recent 1.6.21
Crash Log added.

[crashlog-moonlight-1-6-21.log](https://github.com/user-attachments/files/17395205/crashlog-moonlight-1-6-21.log)

**Comment by Archimedes1986** [comment]
Update:
The same, or very similar issue seems to exists with the KODI App (https://github.com/xbmc/xbmc/issues/25839). 

A user suggested to factory-reset the TV after the update, which in fact solves the problem.

**Comment by gusdom7** [comment]
Performed factory reset. I can confirm is now working, but some issues are present:

Host must be added manually via IP
Using magic remote breaks all Webos UI. It gets really slow, cursor freezes each 2 seconds. I needed to unplug TV to fix.

**Comment by Archimedes1986** [comment]
> Host must be added manually via IP Using magic remote breaks all Webos UI. It gets really slow, cursor freezes each 2 seconds. I needed to unplug TV to fix.

Same here, exactly the same behaviour on my TV.

**Comment by mariotaku** [comment]
This is so strange. I'll be debugging on a device running webOS 9 when I have a chance.

In the meantime, please let me know if you can let me have SSH access!

**Comment by panizzaaaaa** [comment]
Hello, I have the same problem with my LG oled G3 65" since webOS 24 update it does not work, I uninstalled and reinstalled everything including dev mode app

Moonlight directly close at the launch of the application, for information

**Comment by ValoXTech** [comment]
Same issue and same stacktrace. If you need help or ssh access to debug, please let me know.

**Comment by Archimedes1986** [comment]
> This is so strange. I'll be debugging on a device running webOS 9 when I have a chance.

Indeed, i now regret having performed the factory reset, because that very much limited the chances of debugging this. Maybe you get some information from a diff between before and after that reset. Can you give some guidance for users that have not performed the factory reset yet, on how they can gather the required information?

> In the meantime, please let me know if you can let me have SSH access!
As a last resort, i can offer you access, but only if i have you in a conference call simultaneously.

I also think we should  keep an eye on the kodi/xmbc bug thread. I have a gut feeling that the bugs are somehow related. I have btw created a kodi cashlog which might give you additional information.

**Comment by Israfil86** [comment]
I've tried a few things now, I have an LG G3 with the new WebOS24 installed.
At the beginning I wasn't able to start games etc. After a factory reset of the TV I got it to work with the default settings. Then, with all my hopes, I made my personal settings, i.e. resolution, FPS etc. in the Moonlight settings. After I did that, nothing worked again... so I played around with the settings a bit and found that as soon as I change the audio settings in Moonlight from stereo to 5.1, the transmission no longer works and it freezes (Audio-Backend on Auto). But if I leave it on stereo everything works, my other settings are resolution 3840x2160, 120FPS, bitrate 65,000 and HDR on - the rest is unchanged.

Maybe this information will help somehow.

**Comment by mariotaku** [comment]
@ValoXTech That would be awesome! You can find me on [Moonlight's discord server](https://moonlight-stream.org/discord), or reach me via my email. `github` at `mariotaku.me` .

**Comment by mariotaku** [comment]
@Archimedes1986 It's very likely to be related. Both are connected with wayland and EGL.
Once we're able to solve one, we'll be able to solve another.

**Comment by ValoXTech** [comment]
> @ValoXTech That would be awesome! You can find me on [Moonlight's discord server](https://moonlight-stream.org/discord), or reach me via my email. `github` at `mariotaku.me` .

Just wrote you on Discord. In webos-help server 😁

**Comment by mariotaku** [comment]
TL;DR: Unless performing factory reset, moonlight-tv & Kodi wouldn't work and there's not likely to be any fix. The same applies to any other native apps.

---

Thanks to @ValoXTech's help, we did some live debugging, testing all the native programs uses wayland-egl. And it doesn't look good :(
None of the app are able to create a window. We suspect some configuration didn't update properly when they're supposed to after upgrade to webOS 9.

**Comment by BorkforceOne** [comment]
Also having this issue on my C3, happy to provide any information if useful.

**Comment by Kireddoo** [comment]
I have the exact same issues.

After factory reset of TV, reinstalling and everything,  Moonlight on the Lg C3 can not find my SunShine Stream. My phone on the same wifi can find it.

When directly typing in the IPv6 for my PC, Moonlight crashes. Not sure where I can find log-files if there are, but I would be happy to provide.

**Comment by mariotaku** [comment]
@Kireddoo I made a simple mistake that's commented out code for auto-discovery. I'll fix that in the next release #463

**Comment by RedGuilhas** [comment]
Just did a factory reset and installed the latest release. Moonlight opens and I can connect to my PC but it completely bugs out WebOS. Even outside moonlight, WebOS starts running at 1 frame per second or less.

**Comment by Archimedes1986** [comment]
> Just did a factory reset and installed the latest release. Moonlight opens and I can connect to my PC but it completely bugs out WebOS. Even outside moonlight, WebOS starts running at 1 frame per second or less.

On my TV this only happens when I use the magic remote cursor or when I open the settings.
Using just my Xbox controller works fine.

**Comment by panizzaaaaa** [comment]
> Just did a factory reset and installed the latest release. Moonlight opens and I can connect to my PC but it completely bugs out WebOS. Even outside moonlight, WebOS starts running at 1 frame per second or less.

I tought I was the only one who have this kind of problem, I hard restarted my TV 3-4 time, because when I launch Moonlight and even if I close it, the magic remote cursor bug, and even a bluetooth conntect mouse, all the system is slow, from the main page to the settings, is this the same problem you have?

**Comment by RedGuilhas** [comment]
> > Just did a factory reset and installed the latest release. Moonlight opens and I can connect to my PC but it completely bugs out WebOS. Even outside moonlight, WebOS starts running at 1 frame per second or less.
> 
> I tought I was the only one who have this kind of problem, I hard restarted my TV 3-4 time, because when I launch Moonlight and even if I close it, the magic remote cursor bug, and even a bluetooth conntect mouse, all the system is slow, from the main page to the settings, is this the same problem you have?

It's exactly that what's happening to my TV also

**Comment by Bartovguy** [comment]
Also crashing for me after webos 24 update...
happy to provide any info

**Comment by mariotaku** [comment]
After you perform your factory reset, you'll see it stop crashing and for the cursor issue, please move to a separated issue #466.

**Comment by rgomezgo** [comment]
My experience in LG TV C3 75"

Before the factory reset, nothing happened when I tried to open the Moonlight app

After the factory reset. I can open the app and initialize streaming. The moving control bug (reported in other issue) appeared, causing freezing and extreme stuttering, that affects to any other app or TV use until I reboot / unplug the TV. Moreover, even if I don't use the moving control, screen freezes after seconds or few minutes.

**Comment by numiareced** [comment]
@mariotaku hi there!

I'm having the same issue with crashing the Moonlight on opening. It started after upgrade the LG Webos to 23.20.50. 
See the crash dump attached.

[RDXD_moonlight__Linux Crash_ moonlight crashes at location _var_palm_jail_com.limelight.webos_lib_libc.so.6 (__libc_do_syscall+0x5) [0xf74e1ca6]____moonlight.13301.log](https://github.com/user-attachments/files/17531380/RDXD_moonlight__Linux.Crash_.moonlight.crashes.at.location._var_palm_jail_com.limelight.webos_lib_libc.so.6.__libc_do_syscall%2B0x5.0xf74e1ca6.____moonlight.13301.log)

**Comment by Archimedes1986** [comment]
According to @mariotaku this issue cannot be fixed right now (see comment above).
However, in the xmbc/kodi thread someone mentioned recently (https://github.com/xbmc/xbmc/issues/25839#issuecomment-2434774545)  that an uninstall/reinstall of the DevMode App would fix the issue. Maybe this is worth a try?

**Comment by davidspiess** [comment]
I now managed to get it running without factory reset. 
Uninstalling the DevMode App, as @Archimedes1986 suggested, did not the trick for me, but disabling and reenabling the `Dev Mode Status` setting. I also checked the Checkbox on `Clear Cache Data` and restarted the TV, but i am not sure if it did something.
Then i reenabled `Dev Mode Status`. After that the Moonlight App was gone and i had to reinstall it again from the WebOS Dev Manager.
After doing the initial setup and connecting it to the Sunshine Server, it started working again.

**Comment by RedGuilhas** [comment]
> I now managed to get it running without factory reset. Uninstalling the DevMode App, as @Archimedes1986 suggested, did not the trick for me, but disabling and reenabling the `Dev Mode Status` setting. I also checked the Checkbox on `Clear Cache Data` and restarted the TV, but i am not sure if it did something. Then i reenabled `Dev Mode Status`. After that the Moonlight App was gone and i had to reinstall it again from the WebOS Dev Manager. After doing the initial setup and connecting it to the Sunshine Server, it started working again.

Is the TV working properly after using the cursor inside moonlight?

**Comment by davidspiess** [comment]
@RedGuilhas nope i experience the same problem as you with my magic remote.

**Comment by Pjero** [comment]
Same here LG C3. Turned dev mode off and on again, everything works, but when I turn off Moonlight WebOS lags like crazy. The cursor lags, the rest of the buttons don't seem to work at all.

**Comment by rgargente** [comment]
LG C3 here too, same experience as above, but the issue is after I got it to work again performance is much worse than before webos update. It was really smooth before on 4K, 60fps, max bitrate. Now even with minimum settings, half the resolution and half the fps it stutters every few seconds.

**Comment by mariotaku** [comment]
I have added 2 changes in the latest v1.6.23 release.

- It opens a page telling people to re-enable dev mode when it's about to crash because of this issue 
- Fixes graphics related resource leaks and mouse cursor update code

I don't have a real TV running webOS 9 to test but hopefully it will have some improvements.

**Comment by davidspiess** [comment]
I tested the 1.6.23 release on my G3, but WebOS still freezes and stutters after exiting streaming.

**Comment by Pjero** [comment]
1.6.23 still lags for me too, but it is considerably less than before, still unusable tho.

**Comment by mariotaku** [comment]
I'll be closing this issue as it's been addressed. I also posted the issue and the fix LG could do on their forum. Hopefully LG will push an update to their dev mode app to fix this issue for everyone else.

For the cursor issue, please subscribe to #466 for updates.
